packages/playwright/src/index.ts

```diff
@@ -586,12 +593,15 @@ class ArtifactsRecorder {
   }
 
   async didFinishTestFunction() {
-    if (this._testInfo._isFailure() && (this._screenshotMode === 'on' || this._screenshotMode === 'only-on-failure'))
+    const captureScreenshots = this._screenshotMode === 'on' || (this._screenshotMode === 'only-on-failure' && this._testInfo._isFailure());
```

Comment:
This is a semantic change: we start capturing screenshots at the test function end when `screenshot: 'on'`.

Reply:
Let's say this is intended.